src/libraries/Microsoft.PowerFx.Core/Public/Values/NamedValue.cs

```diff
@@ -15,7 +17,24 @@ public class NamedValue
     {
         public string Name { get; }
 
-        public FormulaValue Value { get; }
+        public FormulaValue Value => _value ?? _getFormulaValue().ConfigureAwait(false).GetAwaiter().GetResult();
```

Comment:
>_getFormulaValue() [](http://example.com/codeflow?start=47&length=18)

Should we cache this value instead of calling _getFormulaValue every time Value is accessed in the lazy scenario?

Reply:
Makes sense.